### ISSUE TITLE
API: colWeightedMedians() accepts `w=NULL` whereas weightedMedian() gives an error

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/matrixStats#25]
INCONSISTENCY:
`colWeightedMedians()` accepts `w=NULL` whereas `weightedMedian()` gives an error, e.g.
```r
> colWeightedMedians(X, w=NULL)
[1]  2  5  8 11
> weightedMedian(X[,1], w=NULL)
Error in weightedMedian(X[, 1], w = NULL) :
  Argument 'x' and 'w' are of different lengths: 3 != 0
```

**Comment by Dongcan-Jiang** [comment]
I tried to resolve this issue on dcacc86f20d51e91d7046f7c5ae53a0619b13a2c .

**Comment by HenrikBengtsson** [comment]
@Dongcan-Jiang, thanks for looking into this.  After identifying various issues recently, I realized that there needs to be a decision on whether weights arguments should be (i) mandatory or (ii) optional first. If mandatory, then an error should be thrown.  If optional, then a decision needs to be made whether `NULL`(*) corresponds to optional or not.

I added a new label to the issues that needs some more thoughts before moving forward.

(*) If optional, I'm in favor of using `NULL` over testing for `missing(w)` to specify "weights not given/don't use weights", because then a user can explicitly specify `w=NULL`, which is not possible with `missing()`.
